templates/goshared/msg.go

```diff
@@ -70,6 +85,14 @@ func (e {{ errname . }}) Error() string {
 
 var _ error = {{ errname . }}{}
 
+var _ interface{
+	GetField() string
+	GetReason() string
+	GetKey() bool
+	GetCause() error
+	GetErrorName() string
```

Comment:
So instead of adding `Get*` for each field that is already exported, I think it's reasonable and more idiomatic to break the existing interface of the error types so that the methods are just `Field()`, `Reason()`, etc. I'm not too concerned about a breaking change, but would prefer only one mechanism for accessing this data (either methods or exported fields, but not both).

Reply:
Nice to meet you @rodaine heard a lot of good things about you from @junr03! :) I enjoyed the talk and learning more about some of the things you guys do! I will update the PR with the changes and push it by tomorrow.